app/models/assignment/rotation_generator.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class Assignment < ApplicationRecord
+  class RotationGenerator
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :roster_id, :integer
+    attribute :user_ids
+    attribute :starting_user_id, :integer
+    attribute :start_date, :date
+    attribute :end_date, :date
+
+    validates :roster, presence: true
+    validates :user_ids, presence: true, length: { minimum: 1 }
+    validate :includes_start_user
+    validates :start_date, presence: true
+    validates :end_date, presence: true, comparison: { greater_than_or_equal_to: :start_date }
+
+    def roster
+      @roster ||= Roster.find_by(id: roster_id)
+    end
+
+    def includes_start_user
+      return unless user_ids
+
+      errors.add(:starting_user_id, 'is not included in the list') unless
+        user_ids.include? starting_user_id.to_s
```

Comment:
```suggestion
      return if user_ids.blank? || starting_user_id.blank?
      return if user_ids.include? starting_user_id

      errors.add :starting_user_id, 'is not included in the list'
```
Generally if we're using a modifier (single-line) form of `if` or `unless`, the whole thing should fit on one line.

Then going off that, the more "standard" way of writing these custom validation methods is to filter out our positive conditions first with guard clauses, then add our error at the end.

Reply:
Oh, also, this message should be localized as well